core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -106,6 +107,10 @@ void shouldCheckIfTheUnreleasedCommitsExist() throws IOException, GitAPIExceptio
 
                     LocalDateTime pluginReleaseDate = LocalDateTime.parse(plugin.getReleaseTimestamp().toString(), DateTimeFormatter.ISO_DATE_TIME);
                     assertThat(commitDate, greaterThan(pluginReleaseDate));
+                    assertThat(plugin)
```

Comment:
There are the three steps metioned in [ one of the comments on this issue](https://github.com/jenkins-infra/plugin-health-scoring/pull/298#discussion_r1176498333).

I am trying to implement the first step but my assert case has bugs.

> returns a failure when the last commit date of any file in src/main folder is more recent than the plugin release timestamp

Reply:
What's the bug? What are the current results and expected results?

Reply:
I want to do the following:

> returns a failure when the last commit date of any file in src/main folder is more recent than the plugin release timestamp

I am trying to achieve the above using the following code:

```
assertThat(plugin).usingRecursiveComparison()
                        .comparingOnlyFields("releaseTimestamp")
                        .isEqualTo(ProbeResult.success(HasUnreleasedProductionChangesProbe.KEY, ""));
```
I do not know if I am achieving the desired results.

If I try `... isEqualTo(ProbeResult.error(HasUnreleasedProductionChangesProbe.KEY, ""));` the test case succeeds. 
If I try `... isEqualTo(ProbeResult.success(HasUnreleasedProductionChangesProbe.KEY, ""));` the test case succeeds.

Shouldn't the test case fail for one of the tests?